python/ray/serve/_private/result_wrapper.py

```diff
@@ -0,0 +1,148 @@
+import threading
+import time
+from abc import ABC, abstractmethod
+from typing import Callable, Optional, Union
+
+import ray
+from ray.serve._private.utils import calculate_remaining_timeout
+
+
+class ResultWrapper(ABC):
```

Comment:
```suggestion
class ReplicaResult(ABC):
```

Reply:
not too strong of an opinion, but I find it a bit of an antipattern to call things "wrappers", it's sort of a waste of screen real estate

Reply:
I did feel like the naming was awkward but my brain got stuck on this and couldn't think of a better name, this works better, thanks!